### PR TITLE
Fix #141

### LINKED ISSUE #141: Classroom list data is undefined when joining a classroom
screen shot 2016-03-30 at 13 47 37

Thanks @aliburchard for spotting this.

### PR DESCRIPTION
**Pull request by simoneduca**
@eatyourgreens @shaunanoordin this is ready for review.

### CODE REVIEW

File: src/actions/classrooms.js [hunk 1]
```diff
@@ -131,4 +130,15 @@ export function fetchStudentClassrooms() {
...
+//      dispatch({
+//        type: types.JOIN_CLASSROOM_SUCCESS,
+//        loading: true
+//      });
+//      console.log('JOIN-RESPONSE: ', json.data);
+//      // the following reloads the page causing the classroomlist to refresh
+//      // not sure if there is a better way that uses react-router methods.
+//      window.location.assign('/students/classrooms/')
+//    })
+
```
Comment (eatyourgreens) [inline src/actions/classrooms.js]: Delete all these commented lines?

Comment (simoneduca) [inline src/actions/classrooms.js]: AGH! Done, thanks.

**Comment by shaunanoordin** [comment]
Good news is that Joining a Classroom properly succeeds in adding a student to the classroom via the API. Also, the previously encountered `can't find .length() of undefined` error is gone - the app is stable.

Bad news is that, after joining a Classroom, the list of Classrooms on the sidebar doesn't properly refresh to reflect my newly joined class.

For example:
- I am a student with existing classrooms Apple and Banana.
- I click a link to join classroom Cherry
- The join is successful, but the Classroom list only shows Apple and Banana
- Navigating away and returning to the page will, however, correctly show Apple, Banana and Cherry listed as my classrooms.

screen shot 2016-03-30 at 14 39 50

EDIT: As @eatyourgreens just pointed out, this may be due to the JOIN_CLASSROOM and RECEIVE_STUDENT_CLASSROOM actions being asynchronous, so the list-all-classes doesn't necessarily trigger _after_ the successful join action.

**Comment by shaunanoordin** [comment]
Status: **OK** This branch is stable enough to merge and fixes the app, but we'll need to revisit this later to address the "not showing most recently joined classroom" issue.
Actions: **Merging**

**Pull request merged**